models/users/telemetry.sql

```diff
@@ -0,0 +1,55 @@
+{% set COLUMNS = 'columns' %}
+{{
+  config(
+    materialized = 'incremental',
+    unique_key='uuid',
+    on_schema_change='append_new_columns',
+    indexes=[
+      {COLUMNS: ['uuid'], 'type': 'hash'},
+      {COLUMNS: ['saved_timestamp']},
+      {COLUMNS: ['period_start']},
+      {COLUMNS: ['user_name']},
+      {COLUMNS: ['app_version']},
+    ]
+  )
+}}
+
+SELECT
+  document_metadata.uuid as uuid,
+  document_metadata.saved_timestamp,
+  CONCAT_WS(                                --> Date concatenation from JSON fields, eg. 2021-5-17
+    '-',
+    doc#>>'{metadata,year}',                --> year
+    CASE                                    --> month of the year
+      WHEN
+        string_to_array(substring(doc#>>'{metadata,versions,app}' FROM '(\d+.\d+.\d+)'),'.')::int[] < '{3,8,0}'::int[]
+      THEN
+        (doc#>>'{metadata,month}')::int+1   --> Legacy, months zero-indexed (0 - 11)
+      ELSE
+        (doc#>>'{metadata,month}')::int     --> Month is between 1 - 12
+    END,
+    CASE                                    --> day of the month, else 1
+      WHEN
+        (doc#>>'{metadata,day}') IS NOT NULL
+      THEN
+        doc#>>'{metadata,day}'
+      ELSE
+        '1970-01-01'
+    END
```

Comment:
```suggestion
      ELSE
        '1'
    END
```

This is not correct, if day dosn't exist, we should set it to 1.